### ISSUE TITLE
Blead test failure: t/porting/args_assert.t

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#16851]
Migrated from [rt.perl.org#133866](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=133866) (status was 'resolved')

Searchable as RT133866$

**Comment by p5pRT** [comment]
### From @dur-randir

#### Created by @dur-randir



The following test fails on a fresh blead&#8203;:

% \./perl harness porting/args\_assert\.t
porting/args\_assert\.t \.\. 1/? \# Failed test 2 \-
PERL\_ARGS\_ASSERT\_DUP\_WARNINGS is declared but not used at
porting/args\_assert\.t line 62
porting/args\_assert\.t \.\. Failed 1/2 subtests

Test Summary Report
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
porting/args\_assert\.t \(Wstat&#8203;: 0 Tests&#8203;: 2 Failed&#8203;: 1\)
&nbsp;  Failed test&#8203;:  2
Files=1\, Tests=2\,  0 wallclock secs \( 0\.00 usr  0\.00 sys \+  0\.08 cusr
0\.00 csys =  0\.08 CPU\)
Result&#8203;: FAIL

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.29.9:

Configured by dur-randir at Wed Feb 27 14:51:01 MSK 2019.

Summary of my perl5 (revision 5 version 29 subversion 9) configuration:
  Commit id: c1e47bad34ce1d9c84ed57c9b8978bcbd5a02e98
  Platform:
    osname=darwin
    osvers=13.4.0
    archname=darwin-thread-multi-2level
    uname='darwin isengard.local 13.4.0 darwin kernel version 13.4.0:
mon jan 11 18:17:34 pst 2016; root:xnu-2422.115.15~1release_x86_64
x86_64 '
    config_args='-de -Dusedevel -DDEBUGGING -Dusethreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.9
-DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include -DPERL_USE_SAFE_PUTENV'
    optimize='-O3 -g'
    cppflags='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.9
-DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include'
    ccversion=''
    gccversion='4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -mmacosx-version-min=10.9 -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/lib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib
/usr/lib
    libs=-lpthread -lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-lpthread -ldl -lm -lutil -lc
    libc=
    so=dylib
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=bundle
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags=' '
    lddlflags=' -mmacosx-version-min=10.9 -bundle -undefined
dynamic_lookup -L/usr/local/lib -fstack-protector'



@INC for perl 5.29.9:
    lib
    /usr/local/lib/perl5/site_perl/5.29.9/darwin-thread-multi-2level
    /usr/local/lib/perl5/site_perl/5.29.9
    /usr/local/lib/perl5/5.29.9/darwin-thread-multi-2level
    /usr/local/lib/perl5/5.29.9


Environment for perl 5.29.9:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/dur-randir
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/Users/dur-randir/perlbrew/bin:/Users/dur-randir/perlbrew/perls/perl-5.22.1/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/texbin
    PERLBREW_HOME=/Users/dur-randir/.perlbrew
    PERLBREW_MANPATH=/Users/dur-randir/perlbrew/perls/perl-5.22.1/man
    PERLBREW_PATH=/Users/dur-randir/perlbrew/bin:/Users/dur-randir/perlbrew/perls/perl-5.22.1/bin
    PERLBREW_PERL=perl-5.22.1
    PERLBREW_ROOT=/Users/dur-randir/perlbrew
    PERLBREW_SHELLRC_VERSION=0.84
    PERLBREW_VERSION=0.84
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Wed\, 27 Feb 2019 12&#8203;:14&#8203;:28 GMT\, randir wrote&#8203;:
> This is a bug report for perl from sergey\.aleynikov@&#8203;gmail\.com\,
> generated with the help of perlbug 1\.41 running under perl 5\.29\.9\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> The following test fails on a fresh blead&#8203;:
> 
> % \./perl harness porting/args\_assert\.t
> porting/args\_assert\.t \.\. 1/? \# Failed test 2 \-
> PERL\_ARGS\_ASSERT\_DUP\_WARNINGS is declared but not used at
> porting/args\_assert\.t line 62
> porting/args\_assert\.t \.\. Failed 1/2 subtests
> 
> Test Summary Report
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> porting/args\_assert\.t \(Wstat&#8203;: 0 Tests&#8203;: 2 Failed&#8203;: 1\)
>   Failed test&#8203;:  2
> Files=1\, Tests=2\,  0 wallclock secs \( 0\.00 usr  0\.00 sys \+  0\.08 cusr
> 0\.00 csys =  0\.08 CPU\)
> Result&#8203;: FAIL
>

Confirmed\.  I tried deleting lines 903\-904 of proto\.h\, but while that fixed that test failure it created a new one in t/porting/ss\_dup\.t\.
 

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

Fixed by e14197aa126e873114b4bb8628ed7afa87557457
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'open' to 'resolved'
